dashboard/app/models/pd/workshop.rb

```diff
@@ -822,15 +822,15 @@ def pre_survey_course
 
   # @return an array of tuples, each in the format:
   #   [unit_name, [lesson names]]
-  # Units represent the localized titles for scripts in the Course
-  # Lessons are the stage names for that script (unit) preceded by "Lesson n: "
+  # Units represent the localized titles for units in the Course
+  # Lessons are the lesson names for that unit preceded by "Lesson n: "
   def pre_survey_units_and_lessons
     return nil unless pre_survey?
     pre_survey_course.default_scripts.map do |script|
       unit_name = script.title_for_display
-      stage_names = script.lessons.where(lockable: false).pluck(:name)
-      lesson_names = stage_names.each_with_index.map do |stage_name, i|
-        "Lesson #{i + 1}: #{stage_name}"
+      lesson_names = script.lessons.where(has_lesson_plan: true).or(script.lessons.where(lockable: false)).pluck(:name)
```

Comment:
I think this will work as written, but it could be good to keep in mind that has_lesson_plan is nullable, so we can't reliably check against it being false. lockable is ok because that column is not nullable.

Reply:
Good point. It would probably be good anyway to add that has_lesson_plan should not be null. I can write a separate migration and make sure it goes in before this.

Reply:
sounds good. although since you're checking for `has_lesson_plan: true` here, it should be safe to merge this first.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/38822